templates/rustdoc.hbs

```diff
@@ -9,26 +9,35 @@
     <link rel="search" href="/opensearch.xml" type="application/opensearchdescription+xml" title="Docs.rs">
 </head>
 <body>
-{{> navigation_rustdoc}}
-<div id="rustdoc_body_wrapper" class="{{content.rustdoc_body_class}}" tabindex="-1">
-    {{{content.rustdoc_body}}}
-</div>
-<script type="text/javascript" src="/menu.js?{{cratesfyi_version_safe}}"></script>
+    <div id="page-container">
+        {{> navigation_rustdoc}}
+        <div id="rustdoc_body_wrapper" class="{{content.rustdoc_body_class}}" tabindex="-1">
+            {{{content.rustdoc_body}}}
+        </div>
+    </div>
+    <script type="text/javascript" src="/menu.js?{{cratesfyi_version_safe}}"></script>
 </body>
 <script>
-    var doc_body = document.getElementById("rustdoc_body_wrapper");
-    if(window.location.hash) {
-        var notFirefox = typeof InstallTrigger === 'undefined';
-        if(notFirefox) {
-            var hash = window.location.hash;
-            window.location.hash = "";
-            setTimeout(function () {
-                window.location.hash = hash;
-                doc_body.focus();
-            }, 1);
-        }
-    } else {
-        doc_body.focus();
-    }
+ // restore scroll state across back/forward navigation & give focus
+ // to the scrolling container:
+ (function() {
+     var keyName = function() {
```

Comment:
Why not storing this information once and for all?

Reply:
Not sure what you mean by "once and for all" - the key thing that I want to achieve here is that people can navigate browser tabs back and forth and they get scrolled back to the place that they were when they viewed that page in that tab.

`sessionStorage` is tab-local; that means if you store a scroll position under one key for all pages, that means if you click on a link that takes you to, e.g. a source view in the same tab, this script would scroll you to the position that you were scrolled to on the previous page - not where you meant to go.

So, by keying this on the address that people are on, we restore their scroll position at exactly that address.

This actually reminds me of a thing that *could* occur: You could be on the same location.href in different places in your history & this script would restore your scroll location to the _last_ place you'd visited; this isn't ideal. Maybe I can add the history "depth" number to the storage key - that way, it's one position per page/tab/navigation history item position.

Reply:
No I mean: instead of making a function, why not just store what you do in the function in a variable?

Reply:
Ah, sorry, I misunderstood: That's so that the `hashchange` event for intra-page navigation saves the scroll position under the right key; If I just close over the variable, it would save under the old location.href, but we want to save the new location.href. (:

Reply:
Oh right, my bad! It's used in `savedState`. Sorry, didn't pay attention to that... Then it's solved! :)

Reply:
> You could be on the same location.href in different places in your history & this script would restore your scroll location to the last place you'd visited; this isn't ideal. Maybe I can add the history "depth" number to the storage key - that way, it's one position per page/tab/navigation history item position.

This is a nice improvement but it's certainly not necessary to merge this IMO.